src/main/java/bio/terra/common/flagsmith/FlagsmithService.java

```diff
@@ -61,6 +54,47 @@ public Optional<Boolean> isFeatureEnabled(String feature) {
     return Optional.empty();
   }
 
+  /**
+   * Get feature's value formatted as JSON.
+   *
+   * <p>If Flagsmith is unavailable, feature does not exist or the feature value does not exist,
+   * return {@code Optional.empty()}.
+   */
+  public <T> Optional<T> getFeatureValueJson(String feature, Class<T> clazz) {
+    if (!flagsmithProperties.getEnabled()) {
+      LOGGER.info("Flagsmith is not enabled, use default value");
+      return Optional.empty();
+    }
+    var flagsmith = getFlagsmithClient();
+    try {
+      Flags flags = getWithRetryOnException(flagsmith::getEnvironmentFlags);
+      Object value = flags.getFeatureValue(feature);
+      if (value == null) {
+        return Optional.empty();
+      }
+      return Optional.of(objectMapper.readValue(value.toString(), clazz));
+    } catch (FlagsmithClientError e) {
+      LOGGER.warn("Feature {} not found in {}", feature, flagsmithProperties.getApiUrl(), e);
+    } catch (Exception e) {
```

Comment:
If `getWithRetryOnException` throws `InterruptedException` this will swallow that error.
I think it should be allowed to propagate.